cg/smart_contracts/ClearGDPR.sol

```diff
@@ -31,13 +31,13 @@ contract ClearGDPR {
     mapping(bytes32 => subjectStatus) private subjects;
 
     event Controller_ProcessorsUpdated(address[] newProcessors);
-    event Controller_ConsentGivenTo(bytes32 subjectId, address[] processorsConsented); 
-    event Controller_SubjectDataAccessed(bytes32 subjectId);
-    event Controller_SubjectDataRectified(bytes32 subjectId, uint256 rectificationCount);
-    event Controller_SubjectDataRestricted(bytes32 subjectId, bool directMarketing, bool emailCommunication, bool research);
-    event Controller_SubjectDataObjected(bytes32 subjectId, bool objection);
-    event Controller_SubjectDataErased(bytes32 subjectId); 
-    event Processor_SubjectDataErased(bytes32 subjectId, address processor);
+    event Controller_ConsentGivenTo(bytes32 indexed subjectId, address[] processorsConsented);
+    event Controller_SubjectDataAccessed(bytes32 indexed subjectId);
+    event Controller_SubjectDataRectified(bytes32 indexed subjectId, uint256 rectificationCount);
+    event Controller_SubjectDataRestricted(bytes32 indexed subjectId, bool directMarketing, bool emailCommunication, bool research);
```

Comment:
Did you test these changes in the smart contract? 
Solidity code is very delicate!

Reply:
Yes. After my first change some tests failed(because I tried to make address[] indexed). So I had to revert it back.

Reply:
There's a bunch of traps in Solidity, gotta watch out and test every move!